client/asserts.go

```diff
@@ -36,3 +42,51 @@ func (client *Client) Assert(b []byte) error {
 
 	return nil
 }
+
+// Asserts queries assertions with assertTypeName and matching headers.
```

Comment:
I'm not sure it is worth explaining it here but personally I'm not sure what headers mean here. Are those HTTP headers? Some special assertion headers?